Remove Kafka scaler requirement for CA/cert/key

iterion opened the pull request:
This PR removes some validations from the TriggerAuthentication derived config for kafka. This allows you to use TLS without needing to specify a CA/cert/key. It was tested with an AWS MSK Kafka cluster. I haven't yet completed the checklist, but happy to do so.

Later validations in the shared TLS config allow us to just pass through the empty strings here when not specified.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [x] A PR is opened to update the documentation on https://github.com/kedacore/keda-docs

zroubalik commented:
@iterion do you think it could solve this issue? #1241

zroubalik commented:
@iterion could you please update the relevant section in the docs? https://keda.sh/docs/2.0/scalers/apache-kafka/#authentication-parameters

zroubalik commented:
Fix the tests please

iterion commented:
I don't have enough details on the specific problem in https://github.com/kedacore/keda/issues/1241 to say with certainty. But, I believe Confluent does allow you to enable TLS only for encryption and not for authentication and uses publicly trusted CAs. If all of those are true, it would exactly mirror the issue we faced and this PR would resolve it.

PR for docs (I'll fix the DCO shortly):
https://github.com/kedacore/keda-docs/pull/287

And, I'll fix the tests soon.

zroubalik commented:
@iterion great. Once this PR is ok, I'll merge it and will immediately trigger a release of KEDA 2.0 RC2.

So your PR arrived just in time :)

iterion commented:
I realized through the tests that it's still worth checking for cert missing when keys are specified and the opposite. So, I added a bit of handling for that and updated the test variants. I removed some as they seemed like duplicates, but happy to add them back.

I also updated the docs PR to better note that specifying `cert` requires `key` and I signed my commit. Let me know if there is anything else to address.

zroubalik commented:
> I realized through the tests that it's still worth checking for cert missing when keys are specified and the opposite. So, I added a bit of handling for that and updated the test variants. I removed some as they seemed like duplicates, but happy to add them back.
> 

Makes sense!

> I also updated the docs PR to better note that specifying `cert` requires `key` 

I don't see this note in the PR, have you push it? 😄

iterion commented:
Sorry, yes, just did. I pushed the DCO but forgot to add the changes locally. 😅
